tools/trace_analyzer.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python3
+
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright(c) 2023 Intel Corporation. All rights reserved.
+
+'''
+A tool to analyze SOF trace file.
+'''
+
+import re
+import argparse
+from collections import namedtuple
+
+CPU_CLK = 400000000
+CPU_TIMER = 38400000
+AUDIO_PERIOD = 1000
+TICK_TO_MCPS = CPU_CLK / CPU_TIMER / AUDIO_PERIOD
+
+TraceItem = namedtuple('TraceItem', ['timestamp', 'level', 'context', 'func', 'msg'])
+CompId = namedtuple('CompId', ['ppl_id', 'comp_id'])
+CompPerfInfo = namedtuple('CompPerfInfo', ['timestamp', 'samples', 'period', 'cpu_avg', 'cpu_peak'])
+CompPerfStats = namedtuple('CompPerfStats', ['cpu_avg', 'cpu_peak'])
```

Comment:
Take a look at https://docs.python.org/3/library/dataclasses.html, I started using them in `xtensa-build-zephyr.py` and I've been very happy with them. They may be better for some of these.

Reply:
dataclass, namedtuple and typing.NamedTuple are all factory classes,in our use cases, we don't require the flexibility provided by dataclass, namedtuple and typed.NamedTuple both are good for our use. We could use it for future flexibility, but for now, I will add `frozen=true`

Reply:
I didn't suggest dataclass because it's more flexible. I mentioned it because `namedtuple()` is older and IMHO less readable.

Some interesting stuff at:
https://stackoverflow.com/questions/51671699/data-classes-vs-typing-namedtuple-primary-use-cases

>  We could use it for future flexibility,

So you will not switch to dataclass?

> but for now, I will add `frozen=true`

So you will switch to dataclass?

I'm confused.